### DISCUSSION TITLE
Is there a terraform provider for cue?

### DISCUSSION THREAD

**Discussion by uris77** [cue-lang/cue#1185] [Q&A]
Is there a terraform provider similar to https://github.com/alxrem/terraform-provider-jsonnet ?

We are using `cue` to generate some `json` configuration that in turn create a grafana dashboard for us. But we want to integrate it with `terraform`, so that we only need to specify a couple of parameters that generate templated dashboards. 
Currently we have an external (to terraform) step that generates the json file, that terraform then references. We want to remove this external step.

**Comment by samalba** [comment]
That's the reason we built [dagger](https://dagger.io/), it's basically an execution engine behind cue (the execution happens in containers that are generated automatically) - so you attach any execution step (like scripts) anywhere in the cue code and everything is cached automatically and run in parallel (when possible). The dagger terraform library implements what you describe, removes the external step for calling terraform: https://docs.dagger.io/reference/universe/terraform

Currently in private beta, soon to be open source publicly. But if you fill in the form, I can give you access right away.

**Reply by verdverm** [reply]
Based on the Slack convo, I believe @uris77 is looking to generate JSON from CUE within their existing TF, rather than specifying TF with CUE.

@uris77 can you confirm that running `cue` from TF is the goal?

**Reply by uris77** [reply]
Yes, @verdverm is correct. I implemented something already inspired by the `jsonnet` provider https://registry.terraform.io/providers/alxrem/jsonnet/latest/docs

**Comment by birchb1024** [comment]
You could use the ['universe' terraform provider ](https://github.com/birchb1024/terraform-provider-universe) to call cue directly and return the JSON.  

plug: true
